modules/serverless-apiserversource.adoc

```diff
@@ -0,0 +1,231 @@
++
+Alternatively, you can create a file named `trigger.yaml` and copy the sample code below into it:
++
+----
+apiVersion: eventing.knative.dev/v1alpha1
+kind: Trigger
+metadata:
+  name: testevents-trigger
+  namespace: default
+spec:
+  subscriber:
+    ref:
+      apiVersion: serving.knative.dev/v1
+      kind: Service
+      name: event-display
+----
++
+After you have created the `trigger.yaml` file, apply it by entering the following command:
++
+----
+$ oc apply --filename trigger.yaml
+----
+. Create events by launching a pod in the default namespace.
+
+.Verification steps
+
+You can verify that the Kubernetes events were sent into the Knative eventing system by looking at the message dumper function logs.
+
+You can view the message dumper function logs by entering the following commands:
+----
+$ oc get pods
+$ oc logs -l serving.knative.dev/service=event-display -c user-container
+----
+
+The logs should contain lines similar to the following:
+----
+☁️  cloudevents.Event
+Validation: valid
+Context Attributes,
+  specversion: 1.0
+  type: dev.knative.apiserver.resource.update
+  source: https://10.96.0.1:443
+  subject: /apis/v1/namespaces/default/events/testevents.15dd3050eb1e6f50
+  id: e0447eb7-36b5-443b-9d37-faf4fe5c62f0
+  time: 2019-12-04T14:09:30.917608978Z
+  datacontenttype: application/json
+Data,
+  {
+    "apiVersion": "v1",
+    "count": 1,
+    "eventTime": null,
+    "firstTimestamp": "2019-05-10T23:27:06Z",
+    "involvedObject": {
+      "apiVersion": "v1",
+      "fieldPath": "spec.containers{busybox}",
+      "kind": "Pod",
+      "name": "busybox",
+      "namespace": "default",
+      "resourceVersion": "28987493",
+      "uid": "1efb342a-737b-11e9-a6c5-42010a8a00ed"
+    },
+    "kind": "Event",
+    "lastTimestamp": "2019-05-10T23:27:06Z",
+    "message": "Started container",
+    "metadata": {
+      "creationTimestamp": "2019-05-10T23:27:06Z",
+      "name": "busybox.159d7608e3a3572c",
+      "namespace": "default",
+      "resourceVersion": "506088",
+      "selfLink": "/api/v1/namespaces/default/events/busybox.159d7608e3a3572c",
+      "uid": "2005af47-737b-11e9-a6c5-42010a8a00ed"
+    },
+    "reason": "Started",
+    "reportingComponent": "",
+    "reportingInstance": "",
+    "source": {
+      "component": "kubelet",
+      "host": "gke-knative-auto-cluster-default-pool-23c23c4f-xdj0"
+    },
+    "type": "Normal"
+  }
+----
+
+[id="delete-apiserversource_{context}"]
+== Deleting the ApiServerSource
+
+You can delete the ApiServerSource, trigger, and service created in this guide by entering the following commands:
+----
+$ oc --namespace default delete --filename serviceaccount.yaml
+$ oc --namespace default delete --filename k8s-events.yaml
+$ oc --namespace default delete --filename trigger.yaml
```

Comment:
```suggestion
$ oc delete --filename trigger.yaml
```

Reply:
I would reverse the removal order to first delete the trigger, then the apiserversource and finally the authentication setup to avoid intermediate errors

Reply:
Also you forgot to delete the event-display service here (see the comment below how to do it with `kn`)